src/gui/mainwindow.cpp

```diff
@@ -390,6 +393,36 @@ void MainWindow::quitApp(){
     qApp->quit();
 }
 
+QIcon MainWindow::getIcon() {
+  // on initial launch (first time using ckb) this association checked
+  // will not be present therefore it will be created w/ default value of false
+
+  // If the icon is RGB then the menu should present a monochrome option
+  if ( CkbSettings::get("RGBIcon").toBool() ){
+    changeTrayIconAction->setText("Monochrome Tray Icon");
+    connect(changeTrayIconAction, SIGNAL(triggered()), this, SLOT(changeTrayIconToMonochrome()));
+    return QIcon(":/img/ckb-next.png");
+  }
+  // If the icon is monochrome then the menu should present an RGB option
+  changeTrayIconAction->setText("RGB Tray Icon");
+  connect(changeTrayIconAction, SIGNAL(triggered()), this, SLOT(changeTrayIconToRGB()));
+  return QIcon(":/img/ckb-next-monochrome.png");
+}
+
+void MainWindow::changeTrayIconToMonochrome(){
+    trayIcon->setIcon(QIcon(":/img/ckb-next-monochrome.png"));
+    changeTrayIconAction->setText("RGB Tray Icon");
+    connect(changeTrayIconAction, SIGNAL(triggered()), this, SLOT(changeTrayIconToRGB()));
```

Comment:
You should call disconnect() before calling connect() here, so that the previous signal to changeTrayIconToMonochrome() gets removed.